Sources/NIO/Linux.swift

```diff
@@ -132,5 +136,60 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func contentsOfFile(path: String) throws -> Substring {
+        let fh = try NIOFileHandle(path: path)
+        defer { try? fh.close() }
+        // linux doesn't properly report /sys/fs/cgroup/* files lengths so we use a reasonable limit
+        let buffSize = 1024
+        var buf = ByteBufferAllocator().buffer(capacity: buffSize)
+        try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: buffSize) { ptr in
+            let res = try fh.withUnsafeFileDescriptor { fd -> IOResult<ssize_t> in
+                return try Posix.read(descriptor: fd, pointer: ptr.baseAddress!, size: buffSize)
+            }
+            switch res {
+            case .processed(let n):
+                return n
+            case .wouldBlock:
+                throw LinuxError()
```

Comment:
You can put a

    preconditionFailure("read returned EWOULDBLOCK despite a blocking fd")

here. This will never happen, and if it does, this needs to crash because it would then be programmer error (someone passed `O_NONBLOCK`).